src/libutil/config.hh

```diff
@@ -353,4 +359,23 @@ struct GlobalConfig : public AbstractConfig
 
 extern GlobalConfig globalConfig;
 
+
+struct ExperimentalFeatureSettings : Config {
+
+    Setting<std::set<ExperimentalFeature>> experimentalFeatures{this, {}, "experimental-features",
+        "Experimental Nix features to enable."};
+
+    bool isEnabled(const ExperimentalFeature &) const;
+
+    /* null ptr means no feature, which means there is nothing that could be
+     * disabled, and so the function returns true.
+     */
+    bool isEnabled(const ExperimentalFeature *) const;
+
+    void require(const ExperimentalFeature &) const;
+};
+
+// FIXME: don't use a global variable.
```

Comment:
```suggestion
```

I agree, but I don't think this is an isolated problem.

Reply:
It isn't, but another such global already had this FIXME, so I just copied it for consistency.

Reply:
I must have made a mistake grepping for it. sgtm